Core/Security/Utilities.py

```diff
@@ -40,21 +40,32 @@
 def generateCAFile(location=None):
   """
 
-  Generate a single CA file with all the PEMs
+  Generate/find a single CA file with all the PEMs
 
-  :param str location: we can specify a specific location in CS
-  :return: file cas.pem which contains all certificates
+  :param str location: we can specify a specific CS location
+                       where it's written a directory where to find the CAs and CRLs
+  :return: directory where the file cas.pem which contains all certificates is created
 
   """
   caDir = Locations.getCAsLocation()
-  for fn in (os.path.join(os.path.dirname(caDir), "cas.pem"),
-             os.path.join(os.path.dirname(Locations.getHostCertificateAndKeyLocation(location)[0]), "cas.pem"),
+  if not caDir:
+    return S_ERROR('No CAs dir found')
+
+  if os.path.isfile(os.path.join(os.path.dirname(caDir), "cas.pem")):
+    casFile = os.path.join(os.path.dirname(caDir), "cas.pem")
+    chain = X509Chain.X509Chain()
+    chain.loadChainFromFile(casFile)
+    expired = chain.hasExpired()
+    if expired['OK'] and not expired['Value']:
```

Comment:
cas.pem contains miltiple certificates. Which certificate expiration this check is actually checking ?

Reply:
I honestly don't know what logic would be implemented... my guess is "any" but I didn't check.

Reply:
Why you feel needing to add this check then ?

Reply:
Actually this will check the first certificate in the CAs list only. So, this check here looks irrelevant

Reply:
This check will fail is any of the CA certificates is expired actually. I do not think that this check is relevant. This can result in a situation when one expired CA certificate blocks the system if even it is not used at all.

Reply:
OK I remove it.